src/main/java/com/example/photobucket/services/loginService/LoginService.java

```diff
@@ -0,0 +1,208 @@
+     *
+     * @param username the username of a Login Service user
+     * @throws UserDoesNotExistException if the username does not belong to an existing user
+     * @return true if the user is logged in, otherwise false
+     */
+    public boolean userLoggedIn(String username) {
+        if (!userExists(username)) throw new UserDoesNotExistException("User Does Not Exist");
+        User requestedUser = users.get(username);
+        return requestedUser.loggedIn();
+    }
+
+    /**
+     * Logs out an existing user of the Login Service
+     *
+     * @param username the username of a Login Service user
+     * @throws UserDoesNotExistException if the username does not belong to an existing user
+     */
+    public void logoutUser(String username) {
+        if (!userExists(username)) throw new UserDoesNotExistException("User Does Not Exist");
+        User requestedUser = users.get(username);
+        requestedUser.logOut();
+    }
+
+    /**
+     * Generates a "reset-password" token for a user of the Login Service
+     *
+     * @param username the username of a Login Service user
+     * @throws UserDoesNotExistException if the username does not belong to an existing user
+     */
+    public void generateToken(String username) {
+        if (!userExists(username)) throw new UserDoesNotExistException("User Does Not Exist");
+        User requestedUser = users.get(username);
+        requestedUser.generateToken();
+    }
+
+    /**
+     * Attempts to change the password for a user of the Login Service
+     *
+     * @param username the username of an existing user of the Login Service
+     * @param password the password of an existing user of the Login Service
+     * @param token the "reset-password" token to be used in the password change attempt
+     * @throws UserDoesNotExistException if the username does not belong to an existing user
+     * @throws IllegalChangePasswordTokenException if the "reset-password" token is invalid
+     *                                             (doesn't consist of 6 digits)
+     * @return true is the password change was successful, otherwise false
+     */
+    public boolean changePassword(String username, String token, String password) {
+        if (!userExists(username)) {
+            throw new UserDoesNotExistException("User Does Not Exist");
+        } else if (!token.matches("\\d{6}")) {
+            throw new IllegalChangePasswordTokenException("'reset-password' token needs to consist of 6 digits");
+        }
+        User requestedUser = users.get(username);
+        return requestedUser.changePassword(password, token);
+    }
+
+    /**
+     * A User of the Login Service
+     */
+    private static class User {
+
+        // The username of the user
+        public String username;
+
+        // The password of the user
+        public String password;
+
+        // The log-in state of the user
+        public boolean loggedIn;
```

Comment:
This functionality should be in the login service (for now, we'll move it around when we introduce new components) since username and password are fields that don't change very often as compared to state loggedIn. Also, adding these fields to the User may cause the User class to bloat when other characteristics of the user class are introduced.